### DISCUSSION TITLE
How to make a daily template which respects `foam.openDailyNote.titleFormat`?

### DISCUSSION THREAD

**Discussion by leira** [foambubble/foam#1045] [Q&A] [answered]
There is the `foam.openDailyNote.titleFormat` setting which defines what a daily note title looks like. But if I use a daily template, I have to recreate the `titleFormat` manually. Is there a way, maybe a variable in daily template, to make the title to respect `foam.openDailyNote.titleFormat` setting?

**Comment by riccardoferretti** [comment, marked as answer]
The `foam.openDailyNote` settings came before templates, and basically they are now "deprecated" in favor of using the template. The settings are ignored when a daily note template is found.

Yes I agree there is a bit of inconsistency, the old settings haven't been removed yet for backwards compatibility, but it feels like something that needs to be done at some point.
And unfortunately to follow the VS Code variable pattern, doing the `dateFormat` bit is not easy, except by adding extra properties.

This is correct, you need to recreate the title using the variables defined here (https://code.visualstudio.com/docs/editor/userdefinedsnippets#_snippet-syntax) - note that:
- using `FOAM_DATE_*` the date used will be the one associated to the daily note
- using `CURRENT_*` variables (the standard VS Code day variables) will resolve to today's date

Is there something you can't achieve with templates? or is there some confusion coming from having the two ways of doing daily notes?

We have asked VS Code to provide access to variables outside of editors to improve support for it, you can help by upvoting the following issue: https://github.com/microsoft/vscode/issues/155868

**Reply by leira** [reply]
Thanks for answering my questions here and in Discord too, and providing so much details!
